### PR TITLE
HHH-11135 - TableGenerator.registerExportables should be optional

### PR DESCRIPTION
**Pull request by cfedersp**
Currently TableGenerator only works if registerExportables is called. registerExportables is a narrow use case, and should not be in the critical path. Configure should be the only initialization method that must be called. This patch sets critical class variables in the configure method, so if we never registerExportables, it will still work. Example: if I manually call MyIdGenerator.generate(..) rather than using JPA annotations, registerExportables is never called and the 5.2.2.Final release breaks.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -54,7 +54,7 @@ allprojects {
 }
 
 ext {
-	hibernateTargetVersion = '5.2.2.Final'
+	hibernateTargetVersion = '5.2.2.Fixed'
```
Comment (vladmihalcea) [inline build.gradle]: That needs to be taken out of the PR.

**Comment by vladmihalcea** [comment]
Can you create a Jira issue for this so tht we have more context and be able to track why we needed this change? Thanks.

**Comment by cfedersp** [comment]
Issue 11135 created:
https://hibernate.atlassian.net/browse/HHH-11135

As far as a test goes, I don't know how to get a session, but it would look like this:
@Test
        public void testTableIdGeneration() {
                StandardServiceRegistry serviceRegistry = new StandardServiceRegistryBuilder()
                                .applySetting( AvailableSettings.DIALECT, SequenceDialect.class.getName() )
                                .build();

                try {
					Properties props = new Properties();
                    TableGenerator generator = new TableGenerator();
                    generator.configure( StandardBasicTypes.INTEGER, props, serviceRegistry );

                        
					Object anObjectsExistingId = null;
                    assert(generator.generate((SessionImplementor)session, anObjectsExistingId)!=null);
                } finally {
                    StandardServiceRegistryBuilder.destroy( serviceRegistry );
                }
        }

**Comment by vladmihalcea** [comment]
I asked you a [question on JIRA](https://hibernate.atlassian.net/browse/HHH-11135) related to this issue. I don't understand why you need this change.

**Comment by vladmihalcea** [comment]
The issue has been clarified and a workaround has been suggested.

**Pull request closed without merging**